### PR TITLE
fix(k8s): skip resources without misconfigs

### LINKED ISSUE #7790: bug(k8s): inconsistent results for scan with and without images
## Description
Misconfiguration and Image scans in kubernetes were splitted in #7690, and it brought a nasty bug.

Early the both scans worked in the same function, and misconfigs and vulns from images handled as the same item.
now there are two separated processes. Trivy gets misconfigurations and then appends image vulns.

if Trivy can't receive images (ex there are no permissions), there will be added an empty result and existed misconfig report will be skipped in the report.

**Note**: some kind of this bug was before #7690, but It didn't matter for most cases, because image vulns were adding before misconfigurations.

## Reason
it affects only on `summary` report.
If a resource has no vulns or secret findings and has misconfigs, `index` item will be overridden by the latest result:
https://github.com/aquasecurity/trivy/blob/c0d79fa09e645f3a3dbff878e393b8631fb17b64/pkg/k8s/report/report.go#L90-L96

## Reproduction
I can reproduce this bug in my demo managment k8s cluster.

Scan with `--skip-images`:
```sh
Infra Assessment
┌─────────────┬──────────────────────────────────────────────┬───────────────────┬──────────────────────┬───────────────────┐
│  Namespace  │                   Resource                   │  Vulnerabilities  │  Misconfigurations   │      Secrets      │
│             │                                              ├───┬───┬───┬───┬───┼───┬────┬────┬────┬───┼───┬───┬───┬───┬───┤
│             │                                              │ C │ H │ M │ L │ U │ C │ H  │ M  │ L  │ U │ C │ H │ M │ L │ U │
├─────────────┼──────────────────────────────────────────────┼───┼───┼───┼───┼───┼───┼────┼────┼────┼───┼───┼───┼───┼───┼───┤
│ kube-system │ Service/kube-dns                             │   │   │   │   │   │   │    │ 1  │    │   │   │   │   │   │   │
│ kube-system │ DaemonSet/cilium                             │   │   │   │   │   │   │ 19 │ 28 │ 57 │   │   │   │   │   │   │
│ kube-system │ DaemonSet/kube-proxy                         │   │   │   │   │   │   │ 3  │ 5  │ 9  │   │   │   │   │   │   │
│ kube-system │ Service/hubble-peer                          │   │   │   │   │   │   │    │ 1  │    │   │   │   │   │   │   │
│ kube-system │ Deployment/cilium-operator                   │   │   │   │   │   │   │ 2  │ 4  │ 9  │   │   │   │   │   │   │
│ kube-system │ ConfigMap/cilium-config                      │   │   │   │   │   │   │    │ 1  │    │   │   │   │   │   │   │
│ kube-system │ ConfigMap/extension-apiserver-authentication │   │   │   │   │   │   │    │ 1  │    │   │   │   │   │   │   │
│ kube-system │ Deployment/coredns                           │   │   │   │   │   │   │ 1  │ 4  │ 4  │   │   │   │   │   │   │
└─────────────┴──────────────────────────────────────────────┴───┴───┴───┴───┴───┴───┴────┴────┴────┴───┴───┴───┴───┴───┴───┘
Severities: C=CRITICAL H=HIGH M=MEDIUM L=LOW U=UNKNOWN
```

Scan with unreachable images (without `--skip-images`):
```sh
Infra Assessment
┌─────────────┬──────────────────────────────────────────────┬───────────────────┬───────────────────┬───────────────────┐
│  Namespace  │                   Resource                   │  Vulnerabilities  │ Misconfigurations │      Secrets      │
│             │                                              ├───┬───┬───┬───┬───┼───┬───┬───┬───┬───┼───┬───┬───┬───┬───┤
│             │                                              │ C │ H │ M │ L │ U │ C │ H │ M │ L │ U │ C │ H │ M │ L │ U │
├─────────────┼──────────────────────────────────────────────┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┤
│ kube-system │ ConfigMap/cilium-config                      │   │   │   │   │   │   │   │ 1 │   │   │   │   │   │   │   │
│ kube-system │ Service/hubble-peer                          │   │   │   │   │   │   │   │ 1 │   │   │   │   │   │   │   │
│ kube-system │ ConfigMap/extension-apiserver-authentication │   │   │   │   │   │   │   │ 1 │   │   │   │   │   │   │   │
│ kube-system │ Service/kube-dns                             │   │   │   │   │   │   │   │ 1 │   │   │   │   │   │   │   │
└─────────────┴──────────────────────────────────────────────┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┘
Severities: C=CRITICAL H=HIGH M=MEDIUM L=LOW U=UNKNOWN

```

### PR DESCRIPTION
**Pull request by afdesk**
## Description

This PR fixes overriding of existed misconfigurations found in image scan.

before
```sh
Infra Assessment
┌─────────────┬──────────────────────────────────────────────┬───────────────────┬───────────────────┬───────────────────┐
│  Namespace  │                   Resource                   │  Vulnerabilities  │ Misconfigurations │      Secrets      │
│             │                                              ├───┬───┬───┬───┬───┼───┬───┬───┬───┬───┼───┬───┬───┬───┬───┤
│             │                                              │ C │ H │ M │ L │ U │ C │ H │ M │ L │ U │ C │ H │ M │ L │ U │
├─────────────┼──────────────────────────────────────────────┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┤
│ kube-system │ Service/hubble-peer                          │   │   │   │   │   │   │   │ 1 │   │   │   │   │   │   │   │
│ kube-system │ Service/kube-dns                             │   │   │   │   │   │   │   │ 1 │   │   │   │   │   │   │   │
│ kube-system │ ConfigMap/cilium-config                      │   │   │   │   │   │   │   │ 1 │   │   │   │   │   │   │   │
│ kube-system │ ConfigMap/extension-apiserver-authentication │   │   │   │   │   │   │   │ 1 │   │   │   │   │   │   │   │
└─────────────┴──────────────────────────────────────────────┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┘
Severities: C=CRITICAL H=HIGH M=MEDIUM L=LOW U=UNKNOWN
```

after:
```sh
Infra Assessment
┌─────────────┬──────────────────────────────────────────────┬───────────────────┬──────────────────────┬───────────────────┐
│  Namespace  │                   Resource                   │  Vulnerabilities  │  Misconfigurations   │      Secrets      │
│             │                                              ├───┬───┬───┬───┬───┼───┬────┬────┬────┬───┼───┬───┬───┬───┬───┤
│             │                                              │ C │ H │ M │ L │ U │ C │ H  │ M  │ L  │ U │ C │ H │ M │ L │ U │
├─────────────┼──────────────────────────────────────────────┼───┼───┼───┼───┼───┼───┼────┼────┼────┼───┼───┼───┼───┼───┼───┤
│ kube-system │ Deployment/coredns                           │   │   │   │   │   │   │ 1  │ 4  │ 4  │   │   │   │   │   │   │
│ kube-system │ Service/hubble-peer                          │   │   │   │   │   │   │    │ 1  │    │   │   │   │   │   │   │
│ kube-system │ Service/kube-dns                             │   │   │   │   │   │   │    │ 1  │    │   │   │   │   │   │   │
│ kube-system │ ConfigMap/cilium-config                      │   │   │   │   │   │   │    │ 1  │    │   │   │   │   │   │   │
│ kube-system │ ConfigMap/extension-apiserver-authentication │   │   │   │   │   │   │    │ 1  │    │   │   │   │   │   │   │
│ kube-system │ DaemonSet/cilium                             │   │   │   │   │   │   │ 19 │ 28 │ 57 │   │   │   │   │   │   │
│ kube-system │ DaemonSet/kube-proxy                         │   │   │   │   │   │   │ 3  │ 5  │ 9  │   │   │   │   │   │   │
│ kube-system │ Deployment/cilium-operator                   │   │   │   │   │   │   │ 2  │ 4  │ 9  │   │   │   │   │   │   │
└─────────────┴──────────────────────────────────────────────┴───┴───┴───┴───┴───┴───┴────┴────┴────┴───┴───┴───┴───┴───┴───┘
Severities: C=CRITICAL H=HIGH M=MEDIUM L=LOW U=UNKNOWN
```

## Related issues
- Close #7790 

## Checklist
- [ ] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [ ] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/k8s/report/report.go [hunk 1]
```diff
@@ -90,8 +90,13 @@ func (r Report) consolidate() ConsolidatedReport {
...
 		if vulnerabilitiesOrSecretResource(m) {
 			vulnerabilities = append(vulnerabilities, m)
-		} else {
+		}
+		if misconfigsResource(m) {
+			res, ok := index[m.fullname()]
 			index[m.fullname()] = m
+			if ok {
+				index[m.fullname()].Results[0].Misconfigurations = append(index[m.fullname()].Results[0].Misconfigurations, res.Results[0].Misconfigurations...)
+			}
```
Comment (simar7) [inline pkg/k8s/report/report.go:99]: Could you add a test in `report_test.go` to cover this case?

Comment (afdesk) [reply pkg/k8s/report/report.go:99]: good catch! thanks! 
done https://github.com/aquasecurity/trivy/pull/7797/commits/4bc54345a07542fc3ada52f8df87b3837c8af9f5

**Comment by afdesk** [comment]
@simar7 could you take a look again? thanks!

**Review by simar7 — APPROVED** [review]

**Comment by simar7** [comment]
lgtm @afdesk we just need to resolve merge conflicts prior to merge. Could you update and push that?

**Comment by afdesk** [comment]
@simar7 thanks! sure, I'll fix the conflict

**Pull request merged**